README-PNL.md

```diff
@@ -280,6 +286,46 @@ Essentially, you will need to add the following section in a `server{}` block in
 
 Eventually, you should be able to see DPdash login page in a web browser against your server hostname.
 
+### From non-root URL
+
+If you would like to serve your app from a non-root URL, such as `/dpdash`, you will need to have the `DPDASH_BASE_PATH` environment variable set [on initialization](#optional-variables).
+
+If you did not set this during initialization, you will need to modify the following file:
+```
+${state}/dpdash/configs/dashboard/basePathConfig.js: const basePathConfig = '/dpdash';
+```
+Where `/dpdash` is whatever your new path is. It should begin, but not end, with a `/`.
+
+Again, if you did not set this up during initialization, you must now launch the Singularity shell and run `npm run build`:
+
+```sh
+    singularity shell \
+    -B ${state}:/data \
+    -B ${state}/dpdash/configs/dashboard:/sw/apps/dpdash/server/configs \
+    -B ${state}/dpdash/dist:/sw/apps/dpdash/dist \
+    -B ${state}/dpdash/webpack-build:/sw/apps/dpdash/public/js \
+    -B ${data}:/project_data \
+    ${DPDASH_IMG}
+
+    Singularity> cd /sw/apps/dpdash/
+    Singularity> npm run build
+```
+
+The command to test would now be `curl -L http://0.0.0.0:8000/dpdash`.
+
+For setting up a reverse proxy, the Nginx configuration would look like this:
+
+```cfg
+    location /dpdash {
+            proxy_pass http://127.0.0.1:8000/dpdash;
```

Comment:
The `proxy_pass` usually routes to `http://127.0.0.1:8000/` without the URL again. I know we did it for `/idgen` already but this is making me wonder if we are not doing it the right way. Thoughts?

Here's where I did it for other apps:
https://github.com/pnlbwh/pnlservers/blob/master/nginx/nginx.conf#L56

Reply:
It all depends on whether the app being routed that way uses relative URLs internally. If it does, then it will likely need to be done this way. It may vary according to different backend configurations (like how much work was required for this compared to the small amount in Subject ID Gen).

In short, it is simplest if all relative URLs accessed through the reverse proxy match the relative URLs on `127.0.0.1`. Otherwise, the conflicts must be sorted out on a case by case basis.

Reply:
To clarify this a bit further:

Let's say we make no modifications to the app but route `/dpdash` to `localhost:8000`. All links in the app will still point to e.g. `/admin`, so when we open up `/dpdash` and click on a link to `/admin`, we will get a 404 because there's no Nginx route matching `/admin`, only `/dpdash/admin`.

Let's say we update all links in the app to append `/dpdash`, but still route `/dpdash` to `localhost:8000` in Nginx—in other words, the app itself is still being served from `localhost:8000`. That will mean that the app will be non-functional _without_ the Nginx reverse proxy, because if you just visit `localhost:8000` and follow any link, it will take you to `localhost:8000/dpdash/admin` which would be a 404.

To me, it makes the most sense to allow the app to work with or without the reverse proxy. Such that you could visit `localhost:8000/dpdash` and it would work the same as visiting `hostname/dpdash`.